Src/FluentAssertions/Events/EventAssertions.cs

```diff
@@ -144,13 +144,29 @@ public void NotRaise(string eventName, [StringSyntax("CompositeFormat")] string
     {
         IEventRecording recording = Monitor.GetRecordingFor(PropertyChangedEventName);
 
-        string propertyName = propertyExpression.GetPropertyInfo().Name;
+        string propertyName = propertyExpression?.GetPropertyInfo().Name;
 
-        if (recording.Any(@event => GetAffectedPropertyName(@event) == propertyName))
+        if (string.IsNullOrEmpty(propertyName))
```

Comment:
🤔 I don't think `propertyName` can ever be empty here. It can only be a valid name or `null`